### PR TITLE
add non-conservative form for convection operator

### PR DESCRIPTION
**Pull request by qiauil**
Modifications:

* Add non-conservative form of convection operator: $\mathbf{u}\cdot \nabla \mathbf{u}$. 
* rename the `_multi_channel_eval` func to `_multi_channel_conservative_eval` and remove the 0.5 coefficient.
* Support shifting from conservative form and non-conservative form when initializing the operator.
* Changing the docstring.

### CODE REVIEW

File: exponax/nonlin_fun/_convection.py [hunk 1]
```diff
@@ -18,13 +19,14 @@ def __init__(
...
         single_channel: bool = False,
+        conservative: bool = False,
     ):
         """
         Performs a pseudo-spectral evaluation of the nonlinear convection, e.g.,
         found in the Burgers equation. In 1d and state space, this reads
 
         ```
-            𝒩(u) = - b₁ 1/2 (u²)ₓ
+            𝒩(u) = b₁ u (u)ₓ
```
[the remark this answers is not in the document]
Comment (qiauil) [reply exponax/nonlin_fun/_convection.py:29]: Yes! That was my mistake... all the minus should be kept here.

[the remark this answers is not in the document]
Comment (Ceyron) [reply exponax/nonlin_fun/_convection.py:29]: Great :+1: 
Could you add them again in the PR?

**Comment by Ceyron** [comment]
Hi Qiang,

The docstring is still without the minus. Did you forget to push your changes?

**Comment by qiauil** [comment]
Hi, Felix

I think it should be a GitHub issue as it shows "Checking for the ability to merge automatically…Hang in there while we check the branch’s status." from my side since I pushed the commit. Can you see my other changes, e.g., adding codes for single-channel non-conservative convection?

Sorry to hear you were sick; get well soon :)

**Comment by Ceyron** [comment]
I think it could be related to the fact that I made the repo public on Friday. 
You could try making your fork public and see if that resolves the problem

**Comment by qiauil** [comment]
Hi, Felix,

Good point! I just changed the visibility and pushed again. But it seems not working. I just have a search, and it seems to be a common issue; see https://github.com/orgs/community/discussions/78775, and we might need to add a new pull request if it is still not working later...

**Comment by Ceyron** [comment]
Hmmm, yes, recreating the PR seems to be the easiest solution.

**Comment by Ceyron** [comment]
New PR is #46

**Pull request closed without merging**